packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder, PlaceholderAlignment;
+
+import 'package:flutter/painting.dart';
+
+import 'framework.dart';
+
+/// An immutable widget that is embedded inline within text.
+///
+/// The [child] property is the widget that will be embedded. It is the
+/// widget's responsibility to size itself appropriately, the text will
+/// not enforce any constraints.
+///
+/// The [child] property may contain its own [Widget] children (if applicable),
+/// including [Text] and [RichText] widgets which may include additional
+/// [WidgetSpan]s. Child [Text] and [RichText] widgets will be laid out
+/// independently and occupy a rectangular space in the parent text layout.
+///
+/// [WidgetSpan]s will be ignored when passed into a [TextPainter] directly.
+/// To properly layout and paint the [widget], [WidgetSpan] should be passed
+/// into a [Text.rich] widget.
+///
+/// {@tool sample}
+///
+/// A card with `Hello World!` embedded inline within a TextSpan tree.
+///
+/// ```dart
+/// <TextSpan>[
+///   TextSpan(text: 'Flutter is'),
+///   WidgetSpan(
+///     widget: SizedBox(
+///       width: 120,
+///       height: 50,
+///       child: Card(
+///         child: Center(
+///           child: Text('Hello World!')
+///         )
+///       ),
+///     )
+///   ),
+///   TextSpan(text: 'the best!'),
```

Comment:
Bonus points for adding an image that shows what this looks like.

Reply:
I would feel safer adding images in a separate pass to prevent having to deal with versioning in the images for api docs repo, as this patch plus engine patch are getting large.